### PR TITLE
build dataset push and pull commands

### PR DESCRIPTION
**Pull request by ExtinctWolf83**

### CODE REVIEW

**Comment by 98sean98** [comment]
Should also add `google-auth` and `google-cloud-storage` as `install_requires` in `setup.py`

**Comment by ExtinctWolf83** [comment]
> Should also add `google-auth` and `google-cloud-storage` as `install_requires` in `setup.py`

It has been added
Does not show up in file changes for some reason

**Review by 98sean98 — CHANGES_REQUESTED** [review]
Just a small fix for the `dataset_found` decorator

File: deploifai/context.py [hunk 1]
```diff
@@ -159,3 +159,19 @@ def wrapper(click_context, *args, **kwargs):
...
+    def wrapper(click_context, *args, **kwargs):
+        deploifai_context = click_context.find_object(DeploifaiContextObj)
+        if (
+            deploifai_context.dataset_config is not None
+            and "DATASET" in deploifai_context.dataset_config
+            and "id" in deploifai_context.dataset_config["DATASET"]
+        ):
+            return click_context.invoke(f, *args, **kwargs)
+
+        raise local_config.DeploifaiNotInitialisedError("dataset.cfg not found")
```
[the remark this answers is not in the document]
Comment (ExtinctWolf83) [reply deploifai/context.py:176]: what do you think of `dataset.cfg not found, use deploifai dataset init to initialize a dataset` ?

[the remark this answers is not in the document]
Comment (98sean98) [reply deploifai/context.py:176]: yeah, seems good, this message is printed in the terminal right?

File: deploifai/dataset/pull.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+def pull(context: DeploifaiContextObj, target: str = None):
+    """
+    Download files from the cloud to local.
+    """
+
+    api = context.api
+    dataset_id = context.dataset_config["DATASET"]["id"]
+
+    data = context.api.get_data_storage_info(dataset_id)
+    if len(data) == 0:
```
Comment (98sean98) [inline deploifai/dataset/pull.py:25]: `len(data)` doesn't seem right, `get_data_storage_info` should just return one data storage object, isn't it?

File: deploifai/dataset/push.py [hunk 3]
```diff
@@ -1,18 +1,45 @@
...
+
     api = context.api
-    try:
-        configs = read_config_file()
-    except FileNotFoundError:
-        return
+    dataset_id = context.dataset_config["DATASET"]["id"]
+
+    data = context.api.get_data_storage_info(dataset_id)
+    if len(data) == 0:
```
Comment (98sean98) [inline deploifai/dataset/push.py:25]: same issue here

File: deploifai/dataset/pull.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        click.secho("No dataset information extracted", fg="yellow")
+        click.secho("Please run deploifai dataset init again")
```
Comment (98sean98) [inline deploifai/dataset/pull.py:27]: Probably should just be `Dataset has not been initialized, please run deploifai dataset init`.

File: deploifai/dataset/push.py [hunk 5]
```diff
@@ -1,18 +1,45 @@
...
+        click.secho("No dataset information extracted", fg="yellow")
+        click.secho("Please run deploifai dataset init again")
```
Comment (98sean98) [inline deploifai/dataset/push.py:27]: same issue here

**Review by 98sean98 — CHANGES_REQUESTED** [review]
some changes required

File: deploifai/dataset/pull.py (hunk 2 above)
Comment (ExtinctWolf83) [reply deploifai/dataset/pull.py:25]: Yes
It should be `if data is None instead`.
Has been refactored

File: deploifai/dataset/pull.py (hunk 4 above)
Comment (ExtinctWolf83) [reply deploifai/dataset/pull.py:27]: This only happens if they have tampered with the dataset info or no longer have access to that dataset
So unsure how to handle this

**Review by 98sean98 — APPROVED** [review]

**Pull request merged**